pkg/reconciler/instances/istio/ingress-gateway/deployment_test.go

```diff
@@ -0,0 +1,92 @@
+package ingressgateway_test
+
+import (
+	"context"
+	"testing"
+
+	ingressgateway "github.com/kyma-incubator/reconciler/pkg/reconciler/instances/istio/ingress-gateway"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	istioCR "github.com/kyma-project/istio/operator/api/v1alpha1"
+)
+
+func TestIngressGatewayNeedsRestart(t *testing.T) {
```

Comment:
shouldn't those be separate test cases, similar to what we have been doing in other test files?

Reply:
https://github.com/kyma-incubator/reconciler/pull/1247/commits/f1740337e6becd5d96b241b5b7729381e6181b1d